catalyst/dl/runner/core.py

```diff
@@ -364,23 +378,21 @@ def trace(
             predict_params=predict_params,
         )
 
-        if logdir is not None:
-            filename = utils.get_trace_name(
-                method_name=method_name,
-                mode=mode,
-                requires_grad=requires_grad,
-                opt_level=opt_level,
-            )
-
-            logdir = Path(logdir)
-            output: Path = logdir / "trace"
-            output.mkdir(exist_ok=True, parents=True)
+        # Restore previous state of the model
+        getattr(model, "train" if _is_training else "eval")()
+        utils.set_requires_grad(model, _requires_grad)
+        model.to(_device)
```

Comment:
could we place model state restoring in the end of the script, just before `return traced_model`